### PR TITLE
roundcube: Added new option `maxAttachmentSize` to configure max attachment size

### PR DESCRIPTION
**Pull request by asbachb**
###### Motivation for this change
Previously the attachment size was fixed to 18MB.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
```
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0
remote: Enumerating objects: 899, done.
remote: Counting objects: 100% (504/504), done.
remote: Compressing objects: 100% (113/113), done.
remote: Total 261 (delta 187), reused 192 (delta 133), pack-reused 0
Receiving objects: 100% (261/261), 46.02 KiB | 184.00 KiB/s, done.
Resolving deltas: 100% (187/187), completed with 95 local objects.
From https://github.com/NixOS/nixpkgs
   cda41cf743f..e0f07f9b8d7  master     -> refs/nixpkgs-review/0
$ git worktree add /home/asbachb/.cache/nixpkgs-review/rev-a7fd1e090903ae4ebac97d5f9dd7e3d01467bffb-dirty/nixpkgs e0f07f9b8d737eff6d7f90847f8360afc79fb90a
Preparing worktree (detached HEAD e0f07f9b8d7)
Updating files: 100% (22040/22040), done.
HEAD is now at e0f07f9b8d7 Merge pull request #63165 from CRTified/module/initrd-ovpn
$ nix-env -f /home/asbachb/.cache/nixpkgs-review/rev-a7fd1e090903ae4ebac97d5f9dd7e3d01467bffb-dirty/nixpkgs -qaP --xml --out-path --show-trace
No diff detected, stopping review...
$ git worktree prune
```
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/mail/roundcube.nix [hunk 1]
```diff
@@ -95,6 +95,14 @@ in
       '';
     };
 
+    maxAttachmentSize = mkOption {
+      type = types.int;
+      default = 18;
+      description = ''
+        The maximum attachment size in MB. Note: The value is not 100% accurate since rounducbe uses a percentage of php upload configuration.
```
Comment (Ma27) [inline nixos/modules/services/mail/roundcube.nix:102]: You should probably document the `* 1.3` calculation here to make it clear which value to set.

File: nixos/modules/services/mail/roundcube.nix [hunk 2]
```diff
@@ -95,6 +95,14 @@ in
...
+      '';
```
Comment (Ma27) [inline nixos/modules/services/mail/roundcube.nix:103]: How about adding `apply = x: "${toString (x * 1.3)}M";` here? Then you can always refer to `cfg.maxAttachmentize`.

File: nixos/modules/services/mail/roundcube.nix (hunk 1 above)
Comment (asbachb) [reply nixos/modules/services/mail/roundcube.nix]: I changed the description. Feel free to propose something else if you've some better ideas.

File: nixos/modules/services/mail/roundcube.nix (hunk 2 above)
Comment (asbachb) [reply nixos/modules/services/mail/roundcube.nix:104]: Wasn't aware of that possibility. Changed the code as proposed.

**Comment by Ma27** [comment]
@asbachb lgtm, however I had to fix the manual build since you can only reference options using `opt-services.roundcube.<roundcube-opt>`.

To avoid this in the future, you can test the manual build like this:

```
$ nix build -L -f nixos/release.nix --arg supportedSystems '[ builtins.currentSystem ]' --builders '' manual
```

(I'd recommend to create a shell-alias for this btw)

**Review by Ma27 — APPROVED** [review]
LGTM. Let's wait for ofborg, then I can merge this.

**Comment by asbachb** [comment]
@Ma27 It there's no additional change request. Feel free to squash these commits into a single one.

**Comment by Ma27** [comment]
@asbachb thanks!

**Pull request merged**